trinity/protocol/common/managers.py

```diff
@@ -33,6 +35,26 @@
 TReturn = TypeVar('TReturn')
 
 
+class ResponseTimeTracker:
+
+    def __init__(self) -> None:
+        self.responses: List[Tuple[float, int]] = []
+
+    def get_stats(self) -> str:
+        if not len(self.responses):
+            return 'None'
+        total_msgs = len(self.responses)
+        total_response_time = sum([item[0] for item in self.responses])
+        total_items = sum([item[1] for item in self.responses])
+        avg_rtt = total_response_time / total_msgs
+        per_item_rtt = total_response_time / total_items
+        return 'count=%d, avg_rtt=%.2f, avg_time_per_item=%.5f' % (
+            total_msgs, avg_rtt, per_item_rtt)
+
+    def add(self, time: float, size: int) -> None:
+        self.responses.append((time, size))
```

Comment:
This looks like it has `O(n)` storage/memory footprint.  Unless I'm missing something we need to aggregate these stats using a running average and total count of number of responses.

Reply:
I thought this wouldn't be a problem as it's just a two-tuple for every *response* (not every msg received) and I've rarely seen peer connections lasting more than a few hours. Do you think we need to deal with this now or would a separate, low-priority issue be enough?

Reply:
> Do you think we need to deal with this now or would a separate, low-priority issue be enough?

That works for me.  Can you add it to the next milestone to be sure we address it quickly.

Reply:
I'll actually do it here and save an unnecessary PR roundtrip

Reply:
Done